OpenRA.Game/Orders/UnitOrderGenerator.cs

```diff
@@ -96,7 +96,7 @@ public virtual string GetCursor(World world, CPos cell, int2 worldPixel, MouseIn
 		public virtual bool InputOverridesSelection(World world, int2 xy, MouseInput mi)
 		{
 			var actor = world.ScreenMap.ActorsAtMouse(xy)
-				.Where(a => !a.Actor.IsDead)
+				.Where(a => !a.Actor.IsDead && a.Actor.Info.HasTraitInfo<ISelectableInfo>() && (a.Actor.Owner.IsAlliedWith(world.RenderPlayer) || !world.FogObscures(a.Actor)))
```

Comment:
I know this is just copied, but what is the usecase of selecting allied actors that are covered by fog?

Reply:
The idea here is that we know that allied actors will never be covered by fog, so we can replace the relatively expensive fog check with a relatively cheap is-ally check.

Reply:
I suspect that is what lead to the weird "health bars under fog" bug that dead players can encounter.

Reply:
Do we have an issue for that? I think this may be the first i've heard about it.

Dead players should have the same rules as specs wrt `world.RenderPlayer`, so i'm not sure it can be that simple.

Reply:
Pretty sure I mentioned this on IRC, but I didn't put together a reproduction case yet, so there's probably no issue. This was also more of a general comment that the assumption that all allied actors are revealed is bogus. I don't think we need or should fix this here.